### ISSUE TITLE
Geometry overlap unit test

### ISSUE THREAD

**Issue by kpedro88** [cms-sw/cmssw#31010]
The idea was raised in #30976 that new Phase 2 geometries should only be enabled if they are demonstrated to be free of overlaps.

If possible, it would be ideal to automate the overlap checking using a unit test. This unit test could be based on the Geant4 built-in tool, and it would have to use a moderate tolerance in order to avoid false positives / computational precision issues.

**Comment by kpedro88** [comment]
assign geometry

**Comment by cvuosalo** [comment]
Before an overlap detection unit test is implemented, wouldn't it be necessary to perform a campaign to eliminate the existing Phase 2 overlaps? Is the scope of the presence of overlaps in Phase 2 known?

**Comment by kpedro88** [comment]
Introducing an automated tool might make such a campaign easier. This could be done at first without enforcing the results as a unit test.

**Comment by cvuosalo** [comment]
There has been some evidence that a union of CutTubs can create overlaps. These unions are produced by algorithms in the following files:
```
Geometry/TrackerCommonData/data/PhaseI/pixfwdInnerDiskZminus.xml
Geometry/TrackerCommonData/data/PhaseI/pixfwdInnerDiskZplus.xml
Geometry/TrackerCommonData/data/PhaseI/pixfwdOuterDiskZminus.xml
Geometry/TrackerCommonData/data/PhaseI/pixfwdOuterDiskZplus.xml
```
These files are part of Runs 2 and 3 scenarios.

**Comment by civanch** [comment]
@cvuosalo , is this issue resolved now?

**Comment by cvuosalo** [comment]
I don't think we have a unit test that checks for Phase 2 overlaps. It's not clear to me how it would work. When a developer adds a new Phase 2 scenario, how would the unit test know about it?

**Comment by civanch** [comment]
PrintGeom Info allows overlap check for any Phase-2 scenario. It is not an automatic unit test but it is the unit test, which may be run by hand. There are another approaches using ROOT and/or visualisation. In all cases so far it is some manual work. Should we keep this issue?

**Comment by civanch** [comment]
+1

@kpedro88 , we can close this issue now.

**Comment by kpedro88** [comment]
@civanch is there now a unit test that checks overlaps whenever someone adds a new phase 2 detector geometry?

**Comment by civanch** [comment]
@kpedro88 , we have extended test which test overlaps in whatever geometry, which should be defined as an input parameter. Test has different levels of checks: 1mm, 0.1 mm, 0.01mm. These numbers are also configurable. We do not have automatic test, decides itself that a new geometry appears and it should be tested. We do not have plans to extend overlap testing facility because the current is now fast and robust.

**Comment by kpedro88** [comment]
> We do not have automatic test, decides itself that a new geometry appears and it should be tested.

That is the topic of this issue.
